src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -587,10 +588,22 @@ public static string GetRelativePath(string directory, string fullPath)
                 }
             }
 
-            // add the rest of the full path parts
-            for (int i = index; i < fullPathParts.Length; i++)
+            // add the rest of the full path parts only if there is something to add.
+            if (index < fullPathParts.Length)
             {
-                relativePath = CombinePathsUnchecked(relativePath, fullPathParts[i]);
+                for (int i = index; i < fullPathParts.Length; i++)
+                {
+                    relativePath = CombinePathsUnchecked(relativePath, fullPathParts[i]);
+                }
+            }
+            else
+            {
+                // There was nothing to add from fullPath, we have a path ending
+                // with either "../" or it is empty. Trim trailing separator in former case.
+                if (!string.IsNullOrEmpty(relativePath))
+                {
+                    relativePath = TrimTrailingSeparators(relativePath);
```

Comment:
Could you please simplify this by removing the new if/else/if blocks and just calling this method unconditionally after the original loop? It appears to be cheap to call when the relative path does not have trailing separators.

Reply:
sure.
done that.